client/allocrunner/taskrunner/connect_hook_test.go

```diff
@@ -0,0 +1,246 @@
+package taskrunner
+
+import (
+	"context"
+	"encoding/json"
+	"io/ioutil"
+	"os"
+	"path/filepath"
+	"testing"
+
+	consulapi "github.com/hashicorp/consul/api"
+	consultest "github.com/hashicorp/consul/testutil"
+	"github.com/hashicorp/nomad/client/allocdir"
+	"github.com/hashicorp/nomad/client/allocrunner/interfaces"
+	"github.com/hashicorp/nomad/client/testutil"
+	agentconsul "github.com/hashicorp/nomad/command/agent/consul"
+	"github.com/hashicorp/nomad/helper/testlog"
+	"github.com/hashicorp/nomad/nomad/mock"
+	"github.com/hashicorp/nomad/nomad/structs"
+	"github.com/stretchr/testify/require"
+)
+
+var _ interfaces.TaskPrestartHook = (*connectHook)(nil)
+
+// TestTaskRunner_ConnectHook_Prestart asserts the ConnectHook creates Envoy's
+// bootstrap.json configuration based on Connect proxy sidecars registered for
+// the task.
+func TestTaskRunner_ConnectHook_Ok(t *testing.T) {
+	t.Parallel()
+	testutil.RequireConsul(t)
```

Comment:
This is a nifty helper. Should we add a version as an optional argument here that way this test doesn't fail strangely because of an old version? Doesn't have to be part of this PR

Reply:
I think it's actually more useful this way: it discovered CI was still using Consul 1.0.0!

Leaving it naive also allows us to fairly easily test against old versions and see what works.